### PR TITLE
Feature: support target in individual services/bookmarks

### PR DESCRIPTION
**Pull request by Facinorous-420**
Allows the user to add `target:` to specific individual service and bookmarks, so it will treat specific links similar to the `target:` that can be set in the settings file.

An example service with this PR in mind would be:

```
- Apps:
    - Grafana:
        href: https://example.com/
        icon: grafana
        description: System statistics and graphs
        server: docker2
        target: _self
```

If no target is added in the service/bookmark,  it continues to default to `_blank` like it normally does.

Will need to update the wiki to add this information to the services and bookmark pages, if this is accepted. Thanks.

### CODE REVIEW

File: src/components/bookmarks/item.jsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export default function Item({ bookmark }) {
       <a
         href={bookmark.href}
         title={bookmark.name}
-        target={settings.target ?? "_blank"}
+        target={settings.target ?? bookmark.target ?? "_blank"}
```
Comment (JazzFisch) [inline src/components/bookmarks/item.jsx:14]: Please change from most specific to least specific.  In other words
```jsx
target={bookmark.target ?? settings.target ?? "_blank"}
```

File: src/components/services/item.jsx [hunk 2]
```diff
@@ -71,7 +71,7 @@ export default function Item({ service }) {
             (hasLink ? (
               <a
                 href={service.href}
-                target={settings.target ?? "_blank"}
+                target={settings.target ?? service.target ?? "_blank"}
```
Comment (JazzFisch) [inline src/components/services/item.jsx:74]: Please change from most specific to least specific.  In other words
```jsx
target={service.target ?? settings.target ?? "_blank"}
```

File: src/components/services/item.jsx [hunk 3]
```diff
@@ -84,7 +84,7 @@ export default function Item({ service }) {
           {hasLink ? (
             <a
               href={service.href}
-              target={settings.target ?? "_blank"}
+              target={settings.target ?? service.target ?? "_blank"}
```
Comment (JazzFisch) [inline src/components/services/item.jsx:87]: Please change from most specific to least specific.  In other words
```jsx
target={service.target ?? settings.target ?? "_blank"}
```

**Review by JazzFisch — CHANGES_REQUESTED** [review]
Neat idea, though can you please update the ordering of the evaluations?

**Comment by Facinorous-420** [comment]
> Neat idea, though can you please update the ordering of the evaluations?

My use case which brought this idea in was chaining homepage instances together, meanwhile leaving all other links as open in new tab. 
Thanks for the review! Ive done those changes. Everything should be golden now!

**Comment by Facinorous-420** [comment]
> Neat idea, though can you please update the ordering of the evaluations?

My use case which brought this idea in was chaining homepage instances together, meanwhile leaving all other links as open in new tab. 
Thanks for the review! Ive done those changes. Everything should be golden now!

File: src/components/services/item.jsx [hunk 4]
```diff
@@ -84,7 +84,7 @@ export default function Item({ service }) {
           {hasLink ? (
             <a
               href={service.href}
-              target={settings.target ?? "_blank"}
+              target={service.target ?? settings.target ??  "_blank"}
```
Comment (shamoon) [inline src/components/services/item.jsx:87]:
```suggestion
              target={service.target ?? settings.target ?? "_blank"}
```

**Review by shamoon — CHANGES_REQUESTED** [review]
Sorry just spaces for my brain, feel free to squash these commits =)

**Pull request merged**